src/Menu/_MenuItem.js

```diff
@@ -59,19 +59,26 @@ const MenuItem = ({
                 children.props.className
             );
 
-            const addonChildBefore = addonBefore ? (<span {...addonProps} className={addonBeforeClassnames} />) : null;
-            const addonChildAfter = addonAfter ? (<span {...addonProps} className={addonAfterClassnames} />) : null;
+            const addonChildBefore = addonBefore ? (<span
+                key='addonChildBefore'
+                {...addonProps}
+                className={addonBeforeClassnames} />) : null;
+            const addonChildAfter = addonAfter ? (<span
+                key='addonChildAfter'
+                {...addonProps}
+                className={addonAfterClassnames} />) : null;
 
             return (
-                <React.Fragment>
-                    {React.Children.toArray(children).map(child => {
+                <>
+                    {React.Children.toArray(children).map((child, key) => {
                         return React.cloneElement(child, {
                             'className': childrenClassnames,
+                            key: `menu-item-${key}`,
                             ...urlProps
                         },
-                        [addonChildBefore, (<span className='fd-menu__title'>{child.props.children}</span>), addonChildAfter]);
+                        [addonChildBefore, (<span className='fd-menu__title' key='menuTitle'>{child.props.children}</span>), addonChildAfter]);
```

Comment:
I get that we're adding keys because this stuff is an array,  but it seems really awkward that it is an array to begin with.

and what's with
```js
addonChildBefore, (<span className='fd-menu__title' key='menuTitle'>{child.props.children}</span>), addonChildAfter]
```
being passed in as the `this` argument of `.map`.  

Anyways, I think this block of code is *trying* to be this:
```jsx
return (
    <>
        {addonBefore && <span {...addonProps} className={addonBeforeClassnames} />}
        <span className='fd-menu__title'>
            {React.Children.map(children, child => {
                return React.cloneElement(child, {
                    className: childrenClassnames,
                    ...urlProps
                });
            })}
        </span>
        {addonAfter && <span {...addonProps} className={addonAfterClassnames} />}
    </>
);
```

Reply:
Yeah I honestly didn't think too hard about it 😆  I'll make this change